src/main/java/com/github/valfirst/slf4jtest/TestLoggerAssertions.java

```diff
@@ -0,0 +1,135 @@
+     *
+     * @param level     the level of the log message to look for
+     * @param message   the expected message
+     * @param arguments any optional arguments that may be provided to the log message
+     * @return a {@link TestLoggerAssertions} for chaining
+     */
+
+    public TestLoggerAssertions hasNotLogged(Level level, String message, Object... arguments) {
+        long count = getLogCount(level, logWithMessage(message, arguments));
+        if (count != 0) {
+            if (arguments.length == 0) {
+                failWithMessage("Found %s log with message `%s`, even though we expected not to", level, message);
+            } else {
+                failWithMessage("Found %s log with message `%s` (with arguments), even though we expected not to", level, message);
+            }
+        }
+        return this;
+    }
+
+
+    /**
+     * Verify that a log message, at a specific level, has not been logged by the test logger in the presence of a {@link Throwable}.
+     *
+     * @param level     the level of the log message to look for
+     * @param message   the expected message
+     * @param arguments any optional arguments that may be provided to the log message
+     * @return a {@link TestLoggerAssertions} for chaining
+     */
+    public TestLoggerAssertions hasNotLogged(Throwable throwable, Level level, String message, Object... arguments) {
+        long count = getLogCount(level, logWithMessage(message, Optional.of(throwable), arguments));
+        if (count != 0) {
+            if (arguments.length == 0) {
+                failWithMessage("Found %s log with message `%s` (with throwable), even though we expected not to", level, message);
+            } else {
+                failWithMessage("Found %s log with message `%s` (with throwable and arguments), even though we expected not to", level, message);
+            }
+        }
+        return this;
+    }
+
+    private long getLogCount(Level level, Predicate<LoggingEvent> predicate) {
+        return actual.getLoggingEvents().stream().filter(event -> level.equals(event.getLevel()) && predicate.test(event))
+                .count();
+    }
+
+    private static Predicate<LoggingEvent> logWithMessage(String message, Object... arguments) {
+        return logWithMessage(message, Optional.empty(), arguments);
+    }
+
+    private static Predicate<LoggingEvent> logWithMessage(String message, @SuppressWarnings("OptionalUsedAsFieldOrParameterType") Optional<Throwable> maybeThrowable, Object... arguments) {
+        return event -> {
+            if (!message.equals(event.getMessage())) {
+                return false;
+            }
+
+            if (arguments.length != event.getArguments().size()) {
+                return false;
+            }
+
+            // don't allow arguments to be in a different order
+            if (Collections.indexOfSubList(Arrays.asList(arguments), event.getArguments()) == -1) {
```

Comment:
Doesn't simple `equals` work here?
```suggestion
            if (event.getArguments().equals(Arrays.asList(arguments))) {
```

Reply:
Doh! Yep good point